src/intersections/raytriangle.jl

```diff
@@ -5,6 +5,13 @@
 #=
 Möller, T. & Trumbore, B., 1997.
 (https://www.tandfonline.com/doi/abs/10.1080/10867651.1997.10487468)
+
+Cases
+1. CrossingRayTriangle - middle of ray intersects middle of triangle
+2. CornerOriginTouchingRayTriangle - origin of ray intersects corner of triangle
+3. EdgeOriginTouchingRayTriangle - origin of ray intersects edge of triangle
+4. EdgeTouchingRayTriangle - middle of ray intersects edge of triangle
+5. CornerTouchingRayTriangle - middle of ray intersects corner of triangle
```

Comment:
Can we come up with better names for cases 2 to 5? I know it is hard to come up with these names, but we should try to remove the words Edge and Corner if possible. The prefix Origin is fine because the Ray has an origin, but the triangle only has vertices.

Reply:
Yes, I agree, I was trying to stay as much in convention with what was suggested in previous issues (4 and 5 were suggested in [#308](https://github.com/JuliaGeometry/Meshes.jl/issues/308)). How about:
2. VertexOriginTouchingRayTriangle - origin of ray intersects corner of triangle
3. SegmentOriginTouchingRayTriangle - origin of ray intersects edge of triangle
4. SegmentTouchingRayTriangle - middle of ray intersects edge of triangle
5. VertexTouchingRayTriangle - middle of ray intersects corner of triangle
   
Honestly, I'm not sure what else to call the cases where the edge/segment of thr triangle are involved. I don;t really care so if you have a suggestion we can just go with it.

Reply:
I think the following names are an improvement as they focus on what is being intersected and how. For example, CornerTouching means that the corner of the triangle is being touched and EdgeCrossing means that the edge of the triangle is being crossed:

1. CrossingRayTriangle
2. CornerTouchingRayTriangle
3. EdgeTouchingRayTriangle
4. EdgeCrossingRayTriangle
5. CornerCrossingRayTriangle

Reply:
Makes sense to me! I'll update with this.

Reply:
We already using Touching for other types and it follows nicely with the idea of what is happening besides the geometry that is being touched. In the above convention we have a geometry followed by a verb always.